packages/core/src/components/drawer/drawer.tsx

```diff
@@ -92,19 +93,36 @@ export class Drawer {
     show = show ?? !this.show;
 
     if (show) {
-      this.open.emit();
-      if (this.divElement) {
-        this.slideInRight(this.divElement);
+      const { defaultPrevented } = this.open.emit();
+
+      if (defaultPrevented) {
+        return;
+      }
+
+      this.show = true;
+      if (!this.toggle) {
+        if (this.divElement) {
```

Comment:
This conditional pushes the cognitive complexity above 15 (compare Sonarqube). The null check is performed in the fn, too. Is this really necessary?